### PR TITLE
Leader follower

### PR DESCRIPTION
**Pull request by amit2103**
Pull request for issue  #179 (Leader Follower Pattern). Initial code ready for review.

### CODE REVIEW

**Comment by iluwatar** [comment]
- [ ] Put `under review` badge to the pull request
- [ ] Does the example code implement the pattern correctly and follow good coding practices?
- [ ] Does the example code have enough test coverage?
- [ ] Is the example code commented well enough?
- [ ] Is the example code following JavaDoc conventions?
- [ ] Are the project [coding conventions](https://github.com/iluwatar/java-design-patterns/wiki/3.-Coding-conventions) being followed?
- [ ] Is the class diagram generated correctly?
- [ ] Is the `index.md` implemented correctly so the pattern will show correctly on the web site?

File: leader-follower/src/main/java/com/iluwatar/leaderfollower/App.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import java.io.IOException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * 
+ * Leader Follower is a concurrency pattern where multiple threads can efficiently demultiplex
+ * events and dispatch to event handlers.
```
Comment (iluwatar) [inline leader-follower/src/main/java/com/iluwatar/leaderfollower/App.java]: Please explain a bit more!

File: leader-follower/src/main/java/com/iluwatar/leaderfollower/WorkStation.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
+package com.iluwatar.leaderfollower;
+
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
```
Comment (iluwatar) [inline leader-follower/src/main/java/com/iluwatar/leaderfollower/WorkStation.java]: Add JavaDoc

File: leader-follower/src/main/java/com/iluwatar/leaderfollower/Worker.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
+package com.iluwatar.leaderfollower;
+
+import java.util.List;
+
```
Comment (iluwatar) [inline leader-follower/src/main/java/com/iluwatar/leaderfollower/Worker.java]: Add JavaDoc

**Comment by iluwatar** [comment]
I've done my review. Please comment on this thread when the code is ready for another look.

**Comment by npathai** [comment]
I am a bit late to arrive at the scene. But I will review it as well.

**Comment by iluwatar** [comment]
I will work on this on another branch

**Comment by amit2103** [comment]
@iluwatar  Sorry for the delay...I have this implemented...I will make a new pull request within 2 days. Can you please give me some time

**Comment by iluwatar** [comment]
@amit2103 Ok great, looking forward to the next pull request.

**Comment by amit2103** [comment]
@iluwatar  Thanks

**Comment by amit2103** [comment]
Made the new pull request. Do check

**Pull request closed without merging**